feat(protocol-designer, components): introduce release notes & update react-markdown to v9.0.3

jerader opened the pull request:
closes AUTH-1423

co-authored with @y3rsh 

# Overview

Introduce release notes to Pd, similarly to what we have in the app and robot releases. I also moved `ReleaseNotes` from the `App` to `Components` but it is unclear if Pd wants to display the release notes in the announcements modal or just have the link to Github. It seems like we will eventually want to display it in the modal.

Additionally, `react-markdown` was updated from v9.0.1 to 9.0.3

## Test Plan and Hands on Testing

rebuild your dependencies!

Review the code and make sure it makes sense. The link to the release notes.md doc in the announcements modal isn't quite working since it is based off of a github edge link which the doc doesn't exist in edge yet. it should work after this Pr merges.

But you can check how it looks in the branch here: https://github.com/Opentrons/opentrons/blob/07dca99aaee51dc1c40fa7cb4f6317ddbc739750/protocol-designer/release-notes.md

I made sure it looks good in the app still.

## Changelog

- move `ReleaseNotes` and style sheet out of the app and into components
- add react-markdown as a dependency in `components` and remove it from `app`
- create the `.md` doc in protocol-designer
- plug the link into pd announcement modal

## Risk assessment

low, merging into edge so there is time to improve it. this is just the foundation. also I will need Emily B. to help write the release notes for 8.5 and probably proofread what i have written so far.

y3rsh committed:
feat(pd): release notes as a page

jerader committed:
feat(protocol-designer, components): introduce release notes

closes AUTH-1423

jerader committed:
add back style sheet

jerader commented on components/src/molecules/ReleaseNotes/index.tsx:
```diff
@@ -1,21 +1,20 @@
 import styles from './styles.module.css'
 
 export interface ReleaseNotesProps {
+  isOEMMode: boolean
```
preemptively moved this component to `components` because we might end up displaying it in PD.

jerader commented on protocol-designer/vite.config.mts:
```diff
@@ -29,6 +29,14 @@ export default defineConfig(
           },
         }),
+        {
+          name: 'markdown-loader',
```
this is to prepare for displaying the release notes in the announcement modal in pd

jerader committed:
clean up component

jerader commented:
NOTE: after this PR gets approval from devs, I will follow up with Emily B. to help update the copy in the `.md` doc. I will also followup with design/product to see if we should display the release notes in the PD announcement modal or just keep it accessible via the github link (which is a bit weird - users don't like github).

ncdiehl11 commented on app/src/organisms/ODD/RobotSettingsDashboard/RobotSystemVersionModal.tsx:
```diff
@@ -7,11 +7,11 @@ import {
   Flex,
   OVERFLOW_AUTO,
   SPACING,
+  ReleaseNotes,
```
alphabetize

ncdiehl11 commented on app/src/organisms/ODD/RobotSettingsDashboard/RobotSystemVersionModal.tsx:
```diff
@@ -7,11 +7,11 @@ import {
+  ReleaseNotes,
 } from '@opentrons/components'
-
+import { useIsOEMMode } from '/app/resources/robot-settings'
```
same, can move down

ncdiehl11 commented on components/src/molecules/ReleaseNotes/index.tsx:
```diff
@@ -1,21 +1,20 @@
-import { useIsOEMMode } from '/app/resources/robot-settings/hooks'
+import { LegacyStyledText } from '../../atoms'
+import { Box } from '../../primitives'
+import { COLORS } from '../../helix-design-system'
```
can move up a line

koji commented on protocol-designer/src/organisms/AnnouncementModal/announcements.tsx:
```diff
@@ -449,6 +449,14 @@ export const useAnnouncements = (): Announcement[] => {
+                  <LinkComponent
+                    external
+                    href={RELEASE_NOTES_URL}
+                    textDecoration={TEXT_DECORATION_UNDERLINE}
```
```suggestion
                    textDecoration={TYPOGRAPHY.textDecorationUnderline}
```

ncdiehl11 commented on protocol-designer/src/assets/localization/en/modal.json:
```diff
@@ -66,7 +66,7 @@
       "body5": "Bug fix for Move steps not using the gripper by default",
       "body6": "All protocols now require Opentrons App version 8.2.0+ to run.",
-      "body7": "For more information, see the <link1>Protocol Designer Instruction Manual</link1>."
+      "body7": "For more information, see the <link1>Protocol Designer Instruction Manual</link1> or the <link2>Release Notes."
```
I think this needs a closing tag `</link2>`

jerader replied:
oof good call!

koji commented on components/src/molecules/ReleaseNotes/index.tsx:
```diff
@@ -63,7 +62,7 @@ function UnnumberedListText(props: JSX.IntrinsicAttributes): JSX.Element {
   return (
     <Box
-      borderBottom={`1px solid ${String(COLORS.grey30)}`}
+      borderBottom={`1px solid ${COLORS.grey30}`}
```
👍

ncdiehl11 reviewed (commented):
looks good! left one comment about what I think is a missing tag

koji commented on components/package.json:
```diff
@@ -36,6 +36,7 @@
     "i18next": "^19.8.3",
     "lodash": "4.17.21",
     "react-i18next": "14.0.0",
+    "react-markdown": "9.0.1",
```
it might be good to update the version since the latest version is 9.0.3 which is a patch version.
I think our usage is pretty normal so the update won't have an impact on app.

jerader replied:
cool, will do! thanks!

jerader committed:
address feedback

koji commented:
probably we can add the following to `codecov.yml`

```
 - "protocol-designer/vite.config.mts"
```

jerader committed:
add pd vite config to codecov.yml

jerader committed:
update 8.4.0 release notes

jerader commented:
> probably we can add the following to `codecov.yml`
> 
> ```
>  - "protocol-designer/vite.config.mts"
> ```

@koji i added that but then had a merge conflict. when i rebased, i noticed that all the `ignore` lines in the file are not in edge? i guess i'll just delete all my changes to `codecov.yml`?

jerader committed:
remove changes to codecov.yml

ncdiehl11 reviewed (approved):
this looks good to me
